thekuuzzz/PrefParser.py

```diff
@@ -0,0 +1,139 @@
+        # Returns the name, email and profile name
+        try:
+
+            for d in json_object['account_info']:
+                name = d.get('full_name')
+                email = d.get('email')
+                for c in d['profile']:
+                    profile = c.get('name')
+
+                    return name, email, profile
+        except Exception as e:
+            print('Exception caught: ' + e)
+            raise EnvironmentError.filename
+            sys.exit
+
+    def parseDate():
+        # Parses the last time chrome was updated in ISO 8601 UTC format
+        # Returns the last update date in ISO 8601 UTC
+        try:
+
+            webKitTime = json_object['last_update_date']
+            finalDate = _dateConversion(webKitTime)
+            return finalDate
+        except Exception as e:
+            print("Exception running application, here's what we know")
+            print(e)
+            return 0
+
+    def parseMedia():
+        # Parses the following information from the media_engagement key
+        # 1: List of Websites 2: last modified 3. Media played? 4. visitcount
+        # returns these keys in a dict
+        try:
+            media = {}
+            for c, d in json_object['profile']['content_settings']['exceptions']['media_engagement'].iteritems():
+                media.append(prefs.items(website=c, lastModified=dateConversion(d['last_modified']), playback=mediaPlaybacks, visitCount=visits))
+                media.append(_dateConversion(media.webKitTime))
+                return media
+        except Exception as e:
+            print("Exception running application, here's what we know")
+            print(e)
+            return 0
+
+    def parseExt():
+        # Gets a list of the current installed extensions and last update time
+        # returns a list of extension ID's and the last update time in ISO 8601
+
+        webKitTime = prefs.get(json_object['extensions']['autoupdate']['last_check'])
+        _dateConversion(webKitTime)
+        extensions = {}
+        try:
+            extList = findIn(json_object['extensions']['ids'])
+            for d in extList:
+                extensions.append(d)
+            return extensions, endDate
+        except Exception as e:
+            print("Exception running application, here's what we know")
+            print(e)
+            return 0
+
+    def parsePrint():
+        # Creates a new JSON object and parses it to find currently installed printers
+        # Returns a list of the currently installed printers in chrome
+        printerList = {}
+        newObject = json_object['printing']['print_preview_sticky_settings']['appState']
+        new_json = json.JSONEncoder(newObject, encoding='utf-8', default=None)
+        try:
+            for c in new_json:
+                printerList.append(c.get('selectedDestinationId'))
+                return printerList
+        except Exception as e:
+            raise parseException
```

Comment:
and when you raise an exception you create you should have

```
class Error(Exception):
   """Base error class."""

class ParseException(Error):
   """Error raised when parser fails."""
```

and then you can do `raise ParseException('This is the reason why its raised')`